### PR TITLE
allowed all categories + product modal page

### PR DESCRIPTION
**Pull request by jjuszkiewicz**

### CODE REVIEW

File: src/WL/AppBundle/Lib/Breadcrumbs/BreadcrumbsFactory.php [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+ */
+
+namespace WL\AppBundle\Lib\Breadcrumbs;
+
+
+use WL\AppBundle\Lib\CategoriesAllowed;
+
+class BreadcrumbsFactory
+{
+    public static function newInstance(CategoriesAllowed $categoriesAllowed)
```
Comment (dwilkiewicz) [inline src/WL/AppBundle/Lib/Breadcrumbs/BreadcrumbsFactory.php]: Method name looks like singleton, maybe create, build?

File: src/WL/AppBundle/Lib/Breadcrumbs/BreadcrumbsFactory.php [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+ * Date: 12.10.2016
+ * Time: 15:35
+ */
+
+namespace WL\AppBundle\Lib\Breadcrumbs;
+
+
+use WL\AppBundle\Lib\CategoriesAllowed;
+
+class BreadcrumbsFactory
```
Comment (dwilkiewicz) [inline src/WL/AppBundle/Lib/Breadcrumbs/BreadcrumbsFactory.php]: BreadcrumbsFactory -> BreadcrumbsBuilderFactory?

File: src/WL/AppBundle/Lib/Breadcrumbs/BreadcrumbsBuilder.php [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+
+namespace WL\AppBundle\Lib\Breadcrumbs;
+
+
+use Nokaut\ApiKit\Entity\Category;
+use Nokaut\ApiKit\Ext\Data\Collection\Filters\FiltersAbstract;
+use Nokaut\ApiKit\Ext\Data\Entity\Filter\FilterAbstract;
+use WL\AppBundle\Lib\Type\Breadcrumb;
+
+class BreadcrumbsBuilder
```
Comment (dwilkiewicz) [inline src/WL/AppBundle/Lib/Breadcrumbs/BreadcrumbsBuilder.php]: Common interface with BreadcrumbsAllowedCategoriesBuilder? Class name BreadcrumbsBuilder -> BreadcrumbsAllCategoriesBuilder?

Comment (jjuszkiewicz) [inline src/WL/AppBundle/Lib/Breadcrumbs/BreadcrumbsBuilder.php]: I this a BreadcrumbsAllowedCategoriesBuilder and BreadcrumbsBuilder are best example for inheritance model. Both has public method p(repareBreadcrumbs, appendFilter), but in BreadcrumbsAllowedCategoriesBuilder method prepareBreadcrumbs has to do somethings more.

**Pull request merged**